entity/shaders/solid_stroke.vert

```diff
@@ -13,12 +13,15 @@ uniform StrokeInfo {
 
 in vec2 vertex_position;
 in vec2 vertex_normal;
+in float pen_down;
 
 out vec4 stroke_color;
+out float v_pen_down;
 
 void main() {
   // Push one vertex by the half stroke size along the normal vector.
   vec2 offset = vertex_normal * vec2(stroke_info.size * 0.5);
   gl_Position = frame_info.mvp * vec4(vertex_position + offset, 0.0, 1.0);
   stroke_color = stroke_info.color;
+  v_pen_down = pen_down;
```

Comment:
Rather than take the approach of skipping fragments of in-between path vertices, we may be better served by just not generating the extra geometry in the first place. We might be able to back out of this added param when getting strokes to generate proper merges/caps.
See this [WIP](https://github.com/bdero/impeller/commit/2c83a55f21931dc4970dea1484fee5d4e4708097), which delimits segments in the path and expects content to generate the correct geometry for them (that's the WIP part).

Reply:
Wouldn't that require multiple calls to the shader program with new vertex buffers? I tried asking @chinmaygarde about this and he was saying just draw transparent pixels there instead.

Reply:
We're generating individual triangles right now, not triangle strips or anything like that, so we can append the geometry of all the segments to a single buffer and run a single draw call.

Reply:
Ah nvm, just noticed it is a triangle strip... :)

Reply:
For fills we'll be able to do it that way, those are just triangle vertices. But maybe at some point ones that are without any move to verbs should be triangle strips or fans...